### PR TITLE
Added filter option for filtering out error message 

### LINKED ISSUE #18: Add a filter option
It would be great if we had a configurable filter function that, if defined in the options, gets called BEFORE the options.logger and showMessageBox function get called and returns true or false to decide if the error should pass or not.

That way one could filter out error messages like ERR_CONNECTION_REFUSED that are not catchable

### PR DESCRIPTION
**Pull request by vinodf2f**
Added filter option for filtering out error messages

Fixes #18

### CODE REVIEW

File: filter-error.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
```
Comment (sindresorhus) [inline filter-error.js:1]: The filename should be prefixed with `fixture-`

File: index.js [hunk 2]
```diff
@@ -17,11 +17,15 @@ let installed = false;
 
 let options = {
 	logger: console.error,
-	showDialog: !isDev
+	showDialog: !isDev,
+	filters: [/testError/]
```
Comment (sindresorhus) [inline index.js:21]: ?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
You need to add docs and types.

**Comment by sindresorhus** [comment]
I think the option should be named `filter`.

File: index.js (hunk 2 above)
Comment (vinodf2f) [reply index.js:21]: this is a sample testError regex.
I will remove it from here and add in docs

File: index.d.ts [hunk 3]
```diff
@@ -16,6 +16,14 @@ declare namespace unhandled {
 		*/
 		readonly showDialog?: boolean;
 
+	
+		/**
+	   Filter option that receives an array of regex.
```
Comment (sindresorhus) [inline index.d.ts:21]: This just describes the name and the type, which are already clear. It should describe what it does and its use-cases.

File: index.d.ts [hunk 4]
```diff
@@ -16,6 +16,14 @@ declare namespace unhandled {
...
+	  electron-unhandled would not show the error dialog if error message passed by any of regex.
+        @default []
```
Comment (sindresorhus) [inline index.d.ts:23]: Incorrect indentation.

File: index.d.ts [hunk 5]
```diff
@@ -16,6 +16,14 @@ declare namespace unhandled {
...
+		*/
+		readonly filter?: Array<RegExp>;
```
Comment (sindresorhus) [inline index.d.ts:25]:
```suggestion
		readonly filter?: RegExp[];
```

File: index.d.ts [hunk 6]
```diff
@@ -16,6 +16,14 @@ declare namespace unhandled {
 		*/
 		readonly showDialog?: boolean;
 
+	
+		/**
+	   Filter option that receives an array of regex.
+	  electron-unhandled would not show the error dialog if error message passed by any of regex.
```
Comment (sindresorhus) [inline index.d.ts:22]: This could be better written.

File: index.js [hunk 7]
```diff
@@ -17,11 +17,15 @@ let installed = false;
 
 let options = {
 	logger: console.error,
-	showDialog: !isDev
+	showDialog: !isDev,
+	filter: []
```
Comment (sindresorhus) [inline index.js:21]: This moot as we still have to check whether it's defined.

**Comment by sindresorhus** [comment]
Please spend some more effort on this.

**Comment by sindresorhus** [comment]
Closing as this haven't had any activity in a long while and is not close to being mergeable.

**Pull request closed without merging**